openedx/core/djangoapps/schedules/signals.py

```diff
@@ -56,24 +60,61 @@ def create_schedule(sender, **kwargs):
 
     upgrade_deadline = _calculate_upgrade_deadline(enrollment.course_id, content_availability_date)
 
-    schedule = Schedule.objects.create(
-        enrollment=enrollment,
-        start=content_availability_date,
-        upgrade_deadline=upgrade_deadline
-    )
-
     try:
         get_week_highlights(enrollment.course_id, 1)
         experience_type = ScheduleExperience.EXPERIENCES.course_updates
     except CourseUpdateDoesNotExist:
         experience_type = ScheduleExperience.EXPERIENCES.default
 
+    if (schedule_config.hold_back_ratio is not None
+        and schedule_config.hold_back_ratio > 0
+        and _hold_back_some_users(enrollment, schedule_config, experience_type, upgrade_deadline,
+                                  content_availability_date)
+    ):
+        log.debug('Schedules: User held back from scheduled experiences.')
+        return
+
+    schedule = Schedule.objects.create(
+        enrollment=enrollment,
+        start=content_availability_date,
+        upgrade_deadline=upgrade_deadline
+    )
+
     ScheduleExperience(schedule=schedule, experience_type=experience_type).save()
 
     log.debug('Schedules: created a new schedule starting at %s with an upgrade deadline of %s and experience type: %s',
               content_availability_date, upgrade_deadline, ScheduleExperience.EXPERIENCES[experience_type])
 
 
+def _hold_back_some_users(enrollment, schedule_config, experience_type, upgrade_deadline, content_availability_date):
+    course_id_str = unicode(enrollment.course_id)
+    key = 'hold_back_experience_{0}'.format(course_id_str)
+    try:
+        ExperimentData.objects.get(
+            user=enrollment.user,
+            experiment_id=9,
```

Comment:
Side note: what's a good place to capture the various experiment IDs used in the table?